packages/calcite-components/src/components/functional/InternalLabel.tsx

```diff
@@ -0,0 +1,62 @@
+import { TemplateResult } from "lit-html";
+import { h } from "@arcgis/lumina";
+
+interface InternalLabelProps {
+  labelText?: string;
+  onClick?: () => void;
+  required?: boolean;
+  slot?: any;
+  spaceBottom?: boolean;
+  spaceInlineEnd?: boolean;
+  spaceInlineStart?: boolean;
+  tooltipText?: string;
+}
+
+export const CSS = {
+  container: "internal-label-container",
+  requiredIndicator: "internal-label-required-indicator",
+  spaceBottom: "internal-label-space-bottom",
+  spaceInlineEnd: "internal-label-space-inline-end",
+  spaceInlineStart: "internal-label-space-inline-start",
+  text: "internal-label-text",
+};
+
+const onClickHandler = (onClick?: () => void): void => {
+  if (onClick) {
+    onClick();
+  }
+};
+
+export const InternalLabel = ({
+  labelText,
+  onClick,
+  required,
+  slot,
+  spaceBottom,
+  spaceInlineEnd,
+  spaceInlineStart,
+  tooltipText,
+}: InternalLabelProps): TemplateResult => (
+  <div
+    class={{
+      [CSS.container]: true,
+      [CSS.spaceBottom]: spaceBottom,
+      [CSS.spaceInlineEnd]: spaceInlineEnd,
+      [CSS.spaceInlineStart]: spaceInlineStart,
+    }}
+    onClick={() => onClickHandler(onClick)}
+  >
+    <div class={CSS.text}>
+      {labelText}
+      {required && (
+        <span class={CSS.requiredIndicator} id="required-indicator">
```

Comment:
I wonder if we should hide this element from screen readers using `aria-hidden` since the input will already be marked as required. There's no need to read the asterisk to screen reader users.

More info: https://www.accessibility-developer-guide.com/examples/forms/required/#first-approach-using-aria

Reply:
I wonder if we should use the calcite-icon here instead? https://developers.arcgis.com/calcite-design-system/icons/?icon=asterisk-small&query=asterisk